### PR TITLE
Add developer guide and move build instructions to their own page

### PR DESCRIPTION
**Pull request by ngoldbaum**
* Moves the instructions that are currently in the "Developer" page into a new "Building from Source" page. 
* Adds the developer guide content from #117 to the "Developer" page.
* Link to both pages from the docs homepage.

Here's how it looks on my fork:

[homepage](https://github.com/ngoldbaum/csp/wiki)
[build instructions](https://github.com/ngoldbaum/csp/wiki/98.-Building-From-Source)
[developer guide](https://github.com/ngoldbaum/csp/wiki/99.-Developer)

### CODE REVIEW

File: docs/wiki/99.-Developer.md [hunk 1]
```diff
@@ -1,378 +1,394 @@
...
-chmod 755 Mambaforge-23.3.1-1-Linux-x86_64.sh
-./Mambaforge-23.3.1-1-Linux-x86_64.sh -b -f -u -p csp_venv
-
-. ~/github/csp_venv/etc/profile.d/conda.sh
+$ cd csp
+$ git remote rename origin upstream
+# for SSH authentication
+$ git remote add origin git@github.com:<username>/csp.git
+# for HTTP authentication
+$ git remote add origin https://github.com/ngoldbaum/csp.git
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:32]: s/ngoldbaum/<username>

File: docs/wiki/99.-Developer.md [hunk 2]
```diff
@@ -1,378 +1,394 @@
...
-conda config --set channel_priority strict
-conda activate base
-```
+If you have not already configured `ssh` access to GitHub, you can find
+instructions to do so
+[here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh),
+including instructions to create an SSH key if you have not done
+so. Authenticating with SSH is usually the easiest route. If you are
+working in an environment that does not allow SSH connections to GitHub,
+you can look into [configuring a hardware
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:43]: PAT works too `https://<username>:<PAT>@github.com/...` I would just link to github docs

File: docs/wiki/99.-Developer.md [hunk 3]
```diff
@@ -1,378 +1,394 @@
...
-# sudo apt-get install -y automake bison cmake curl flex ninja-build tar unzip zip
 
-# for g++
-sudo apt install build-essential
-```
+3. Switch to the PR branch
 
-**Fedora/RedHat/Centos/Rocky/Alma/etc**
+    ```bash
+    $ git switch LOCAL_BRANCH_NAME
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:115]: `$` not needed

File: docs/wiki/99.-Developer.md [hunk 4]
```diff
@@ -1,378 +1,394 @@
...
 
 ```bash
-# for vcpkg
-make dependencies-mac
-# or
-# brew install bison cmake flex make ninja
-
-# for g++
-brew install gcc
+$ git push upstream LOCAL_BRANCH_NAME
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:133]: `$`

File: docs/wiki/99.-Developer.md [hunk 5]
```diff
@@ -1,378 +1,394 @@
...
 
-On `aarch64` Linux the VCPKG_FORCE_SYSTEM_BINARIES environment variable must be set before running `make build`:
+1. Ensure your local clone of `csp` is synced up with GitHub, including
+   any tags that have been pushed since you last synced:
 
-```bash
-VCPKG_FORCE_SYSTEM_BINARIES=1 make build
-```
+    ```bash
+    $ git pull upstream main --tags
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:214]: `$`

File: docs/wiki/99.-Developer.md [hunk 6]
```diff
@@ -1,378 +1,394 @@
...
+    repository. Using a name like `release/v0.3.4` should avoid any
+    conflicts and make it clear what the branch is for.
 
-| Language | Linter | Autoformatter | Description |
-| :------- | :----- | :------------ | :---------- |
-| C++      | `clang-format` | `clang-format` | Style |
-| Python   | `ruff`         | `ruff` | Style |
-| Python   | `isort`         | `isort` | Imports |
+    ```bash
+    $ git checkout -b relese/v0.x.x
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:224]: `$`

File: docs/wiki/99.-Developer.md [hunk 7]
```diff
@@ -1,378 +1,394 @@
...
+    update the codebase to use the next bugfix version number if you do:
 
-**C++ Autoformatting**
-```bash
-make fix-cpp
-# or
-# clang-format -i -style=file `find ./cpp/ -name "*.*pp"`
-```
+    ```bash
+    $ make patch
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:231]: `$`

File: docs/wiki/99.-Developer.md [hunk 8]
```diff
@@ -1,378 +1,394 @@
...
 
-On Debian/Ubuntu Linux, run
+    Use the version number `bump2version` generated and make sure the
+    tag name begins with a `v`.
 
-```bash
-sudo apt-get install -y graphviz
-```
+    ```bash
+    $ git tag v0.2.0
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:258]: `$`

File: docs/wiki/99.-Developer.md [hunk 9]
```diff
@@ -1,378 +1,394 @@
...
-# or python -m pytest -v csp/tests --junitxml=junit.xml --cov=csp --cov-report xml --cov-report html --cov-branch --cov-fail-under=80 --cov-report term-missing
-```
+If this is your first release, you will need an account on pypi.org and
+your account will need to be added as a maintainer to the `csp` project
+on pypi. You will also need to have two factor authentication enabled on
+your PyPI account.
 
-Adapters might rely on external services. For example, the `kafka` adapter requires a cluster against which to test. We rely on [docker compose](https://docs.docker.com/compose/) for these services. We store docker compose files in the `ci` directory, and convenience `Makefile` commands are available:
+Once that is set up, navigate to the API token page in your PyPI
+settings and generate an API token scoped to the `csp` project. *Do not*
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:289]: s/*/**

File: docs/wiki/99.-Developer.md [hunk 10]
```diff
@@ -1,378 +1,394 @@
...
 
-If your work is still in-progress, open a [draft pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests). Otherwise, open a normal pull request. It might take a few days for a maintainer to review and provide feedback, so please be patient. If a maintainer asks for changes, please make said changes and squash your commits if necessary. If everything looks good to go, a maintainer will approve and merge your changes for inclusion in the next release.
+##### Download release artifacts from github actions
 
-Please note that non substantive changes, large changes without prior discussion, etc, are not accepted and pull requests may be closed.
+Make sure you are in the root of the `csp` repository and execute the
+following commands.
 
+```bash
+$ rm -r dist
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:323]: `$`

File: docs/wiki/99.-Developer.md [hunk 11]
```diff
@@ -1,378 +1,394 @@
...
+This happens as part of the CI so this should only be a double-check.
 
-For example, you may need to `brew install pkg-config`.
 
-### Building thrift:arm64-osx/thrift:x64-osx failed
+##### Optionally upload to testpypi to test "pip install"
 
 ```
-Thrift requires bison > 2.5, but the default `/usr/bin/bison` is version 2.3.
+$ twine upload --repository testpypi dist/*
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:347]: `$`

File: docs/wiki/99.-Developer.md [hunk 12]
```diff
@@ -1,378 +1,394 @@
...
 
-### CMake was unable to find a build program corresponding to "Unix Makefiles".
-
-Complete error message:
-
-```bash
-CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
-CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
+```
+$ pip install --index-url https://test.pypi.org/simple --extra-index-url https://pypi.org/simple csp
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:354]: `$`

File: docs/wiki/99.-Developer.md [hunk 13]
```diff
@@ -1,378 +1,394 @@
...
-```
+##### Upload to pypi
 
-`install-arm64-osx-dbg-out.log`:
+If you are sure the release is ready, you can upload to pypi like so:
 
 ```bash
-ld: Assertion failed: (resultIndex < sectData.atoms.size()), function findAtom, file Relocations.cpp, line 1336.
-collect2: error: ld returned 1 exit status
+$ twine upload --repository csp dist/*`
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:365]: `$`

File: docs/wiki/99.-Developer.md [hunk 14]
```diff
@@ -1,378 +1,394 @@
...
-```bash
-CXX=/opt/homebrew/bin/g++-13 CXXFLAGS='-Wl,-ld_classic' make build
 ```
-
-Intel:
-
-```bash
-CXX=/usr/local/opt/gcc/bin/g++-13 CXXFLAGS='-Wl,-ld_classic' make build
+View at:
+https://pypi.org/project/csp/<version number>/
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:375]: enclose in \`

File: docs/wiki/Home.md [hunk 15]
```diff
@@ -22,6 +22,20 @@ adapters or in realtime mode using realtime input adapters.
 - [8. Profiler](https://github.com/Point72/csp/wiki/8.-Profiler)
 - [9. Caching](https://github.com/Point72/csp/wiki/9.-Caching)
 
+# Installation
+
+We ship binary wheels to install `csp`  on MacOS and Linux via `pip`:
+
+```
+$ pip install csp
```
Comment (timkpaine) [inline docs/wiki/Home.md:30]: change to \`\`\`bash, remove `$`

**Review by timkpaine — CHANGES_REQUESTED** [review]

File: docs/wiki/99.-Developer.md (hunk 2 above)
Comment (ngoldbaum) [reply docs/wiki/99.-Developer.md:43]: Thanks, TIL!

File: docs/wiki/99.-Developer.md (hunk 14 above)
Comment (ngoldbaum) [reply docs/wiki/99.-Developer.md:375]: Sorry, what do you mean here?

Comment (timkpaine) [reply docs/wiki/99.-Developer.md:376]: nvm its already in \`\`\`!

File: docs/wiki/99.-Developer.md [hunk 16]
```diff
@@ -1,378 +1,395 @@
...
-Tweak this script to create a conda environment, install the build dependencies, build, and install a development version of `csp` into the environment.
+Next, you should set some names for the `git` remotes corresponding to
+main Point72 repository and your fork. If you started with a clone of
+the main `Point72` repository, you could do something like:
 
-## Install conda
 ```bash
-mkdir ~/github
-cd ~/github
+$ cd csp
```
Comment (timkpaine) [inline docs/wiki/99.-Developer.md:27]: `$` here as well, i missed on first pass

**Review by timkpaine — CHANGES_REQUESTED** [review]

**Review by timkpaine — APPROVED** [review]

**Pull request merged**